src/xbot/common/math/PIDPropertyManager.java

```diff
@@ -70,14 +85,32 @@ public double getErrorThreshold() {
     }
     
     public void setErrorThreshold(double errorThreshold) {
+        assertionManager.assertTrue(errorThreshold >= 0, "Thresholds won't work if they are negative!");
         propErrorThreshold.set(errorThreshold);
     }
     
     public double getDerivativeThreshold() {
         return propDerivativeThreshold.get();
     }
     
-    public void setDerivativeThreshold(double errorThreshold) {
-        propDerivativeThreshold.set(errorThreshold);
-    }    
+    public void setDerivativeThreshold(double derivativeThreshold) {
+        assertionManager.assertTrue(derivativeThreshold >= 0, "Thresholds won't work if they are negative!");
```

Comment:
If it's a negative number and we are running on-robot, I think we should take the absolute value so it's at least sane.